pandas/core/dtypes/cast.py

```diff
@@ -159,29 +167,19 @@ def maybe_downcast_to_dtype(result, dtype):
         return converted
 
     # a datetimelike
-    # GH12821, iNaT is casted to float
+    # GH12821, iNaT is cast to float
     if dtype.kind in ["M", "m"] and result.dtype.kind in ["i", "f"]:
-        if hasattr(dtype, "tz"):
+
+        if is_datetime_or_timedelta_dtype(dtype):
```

Comment:
i think this code here can be greatly simplified, see what the incoming inputs are and i bet you can adjust it

Reply:
IOW the way this is written L173 is *always* hit

Reply:
tried this. There are three tests that use the else block:
```
FAILED pandas/tests/dtypes/cast/test_downcast.py::test_datetime_with_timezone[True]
FAILED pandas/tests/frame/methods/test_combine_first.py::TestDataFrameCombineFirst::test_combine_first_timezone
FAILED pandas/tests/groupby/test_function.py::test_arg_passthru - TypeError: ...
```
Reverting

Reply:
and you have L177. so i would revert to the original code here as well (except for the period change);

Reply:
Done